internal/controller/nodedisruption_controller_test.go

```diff
@@ -0,0 +1,154 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+/*
+Ideally, we should have one `<kind>_controller_test.go` for each controller scaffolded and called in the `suite_test.go`.
+So, let's write our example test for the CronJob controller (`cronjob_controller_test.go.`)
+*/
+
+/*
+As usual, we start with the necessary imports. We also define some utility variables.
+*/
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	nodedisruptionv1alpha1 "github.com/criteo/node-disruption-controller/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = Describe("NodeDisruption controller", func() {
+
+	// Define utility constants for object names and testing timeouts/durations and intervals.
+	const (
+		NDName      = "test-nodedisruption"
+		NDNamespace = "default"
+
+		timeout  = time.Second * 10
+		duration = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	// TODO: breakdown in multiple pieces
+	Context("TODO", func() {
```

Comment:
You could use `ginkgo`'s constructs to create reusable/more readable tests: `BeforeEach` to create create supporting objects, and `Context` to split tests (with/without disruption budget) etc.